apis/vshn/postgres/v1/backup_types.go

```diff
@@ -0,0 +1,145 @@
+
+	// Metadata holds field path name metadata
+	Metadata = "metadata"
+	// Status holds field path name status
+	Status = "status"
+	// Process holds field path name process
+	Process = "process"
+	// BackupInformation holds field path name backupInformation
+	BackupInformation = "backupInformation"
+	// DatabaseInstance holds field path name databaseInstance
+	DatabaseInstance = "databaseInstance"
+	// InstanceNamespace holds field path name instanceNamespace
+	InstanceNamespace = "instanceNamespace"
+)
+
+// VSHNPostgreSQLName represents the name of a VSHNPostgreSQL
+type VSHNPostgreSQLName string
+
+// VSHNPostgreSQLNamespace represents the namespace of a VSHNPostgreSQL
+type VSHNPostgreSQLNamespace string
+
+// +kubebuilder:object:root=true
+
+// VSHNPostgresBackup defines VSHN managed PostgreSQL backups
+type VSHNPostgresBackup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VSHNPostgresBackupSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// VSHNPostgresBackupList defines a list of VSHNPostgresBackup
+type VSHNPostgresBackupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []VSHNPostgresBackup `json:"items"`
+}
+
+// VSHNPostgresBackupSpec defines the desired state of VSHNPostgresBackup
+type VSHNPostgresBackupSpec map[string]runtime.RawExtension
+
+// VSHNPostgresBackup needs to implement the builder resource interface
+var _ resource.Object = &VSHNPostgresBackup{}
+
+func (in *VSHNPostgresBackup) GetObjectMeta() *metav1.ObjectMeta {
+	return &in.ObjectMeta
+}
+
+func (in *VSHNPostgresBackup) NamespaceScoped() bool {
+	return true
+}
+
+func (in *VSHNPostgresBackup) New() runtime.Object {
+	return &VSHNPostgresBackup{}
+}
+
+func (in *VSHNPostgresBackup) NewList() runtime.Object {
+	return &VSHNPostgresBackupList{}
+}
+
+// GetGroupVersionResource returns the GroupVersionResource for this resource.
+// The resource should be the all lowercase and pluralized kind
+func (in *VSHNPostgresBackup) GetGroupVersionResource() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    apis.GroupVersion.Group,
+		Version:  apis.GroupVersion.Version,
+		Resource: "vshnpostgresbackups",
+	}
+}
+
+// IsStorageVersion returns true if the object is also the internal version -- i.e. is the type defined for the API group or an alias to this object.
+// If false, the resource is expected to implement MultiVersionObject interface.
+func (in *VSHNPostgresBackup) IsStorageVersion() bool {
+	return true
+}
+
+var _ resource.ObjectList = &VSHNPostgresBackupList{}
+
+func (in *VSHNPostgresBackupList) GetListMeta() *metav1.ListMeta {
+	return &in.ListMeta
+}
+
+func init() {
+	apis.SchemeBuilder.Register(&VSHNPostgresBackup{}, &VSHNPostgresBackupList{})
+}
+
+func New() *VSHNPostgresBackup {
+	return &VSHNPostgresBackup{}
+}
+
+// BackupInfo holds necessary data for VSHNPostgresBackup
+type BackupInfo struct {
```

Comment:
The name here seems a bit generic. Or will this also be usable for other backup types in the future? Like Redis?

Reply:
No, definitely not. We know naming is hard so I am open to suggestions

Reply:
maybe `PgBackupInfo`?

Reply:
I changed to `SGBackupInfo`. I think that's the best way to move forward.